include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,470 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BATTERY
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+#define ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <device.h>
+
+/*
+ * Sometimes we have hardware to detect battery present, sometimes we have to
+ * wait until we've been able to talk to the battery.
+ */
+enum battery_present {
+	BATTERY_PRESENCE_UNINITIALIZED = -1,
+	BATTERY_PRESENCE_NO = 0,
+	BATTERY_PRESENCE_YES = 1,
+	BATTERY_PRESENCE_UNKNOWN,
+};
+
+enum battery_cutoff_states {
+	BATTERY_CUTOFF_STATE_NORMAL = 0,
+	BATTERY_CUTOFF_STATE_CUT_OFF,
+	BATTERY_CUTOFF_STATE_PENDING,
+};
+
+enum battery_disconnect_state {
+	BATTERY_DISCONNECTED = 0,
+	BATTERY_NOT_DISCONNECTED,
+	BATTERY_DISCONNECT_ERROR,
+};
+
+struct battery_static_info {
+	uint16_t design_capacity;
+	uint16_t design_voltage;
+	uint32_t cycle_count;
+	uint16_t serial[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+	char manufacturer[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+	char device_name[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+	char chemistry[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+	uint8_t vendor_param[CONFIG_BATTERY_STATIC_INFO_FIELD_LENGTH];
+};
+
+/* Battery parameters */
+struct batt_params {
```

Comment:
(see my comment on current 'battery_info')
Could this perhaps be renamed 'battery_health' ?  'params' is such a general term.